plugin/storage/cassandra/spanstore/operation_names.go

```diff
@@ -126,36 +128,35 @@ func NewOperationNamesStorage(
 }
 
 // Write saves Operation and Service name tuples
-func (s *OperationNamesStorage) Write(serviceName string, operationName string) error {
-	var err error
-	//TODO: take spanKind from args
-	spanKind := ""
-
-	if inCache := checkWriteCache(serviceName+"|"+spanKind+"|"+operationName, s.operationNames, s.writeCacheTTL); !inCache {
-		q := s.table.createWriteQuery(s.session.Query(s.table.insertStmt), serviceName, spanKind, operationName)
-		err2 := s.metrics.Exec(q, s.logger)
-		if err2 != nil {
-			err = err2
+func (s *OperationNamesStorage) Write(operation dbmodel.Operation) error {
+	key := fmt.Sprintf("%s|%s|%s",
+		operation.ServiceName,
+		operation.SpanKind,
+		operation.OperationName,
+	)
+	if inCache := checkWriteCache(key, s.operationNames, s.writeCacheTTL); !inCache {
+		q := s.table.createWriteQuery(
+			s.session.Query(s.table.insertStmt),
+			operation.ServiceName,
+			operation.SpanKind,
+			operation.OperationName,
+		)
+		err := s.metrics.Exec(q, s.logger)
+		if err != nil {
+			return err
 		}
 	}
-	return err
+	return nil
 }
 
 // GetOperations returns all operations for a specific service traced by Jaeger
-func (s *OperationNamesStorage) GetOperations(service string) ([]string, error) {
-	operations, err := s.table.getOperations(s, &spanstore.OperationQueryParameters{
-		ServiceName: service,
+func (s *OperationNamesStorage) GetOperations(
+	query *spanstore.OperationQueryParameters,
+) ([]*spanstore.Operation, error) {
+	return s.table.getOperations(s, &spanstore.OperationQueryParameters{
```

Comment:
you can pass `query` to this function directly

Reply:
Good catch! I got lost during refactors between branches...

Reply:
that's because you're using a convoluted process for stacking diffs, instead of the usual chained branches.

Reply:
This branch is based on branch #1921-2, if that's what you mean by chained branches. 
But we need to merge the PRs in reverse order into chained branches. E.g. PR3 merge to branch 2 then PR2 merge into branch 1, then PR1 into master.
Every time we make changes in branch 1, we will need to propagate changes from branch 1 all the way to branch N.

I am not sure if I should open the PR for endpoint changes against this branch.